src/util/createServerRootMixin.js

```diff
@@ -40,31 +61,56 @@ function searchOnlyWithDerivedHelpers(helper) {
   });
 }
 
-function defaultCloneComponent(componentInstance) {
+function defaultCloneComponent(componentInstance, { mixins = [] } = {}) {
   const options = {
     serverPrefetch: undefined,
     fetch: undefined,
     _base: undefined,
     name: 'ais-ssr-root-component',
-    // copy over global Vue APIs
-    router: componentInstance.$router,
-    store: componentInstance.$store,
   };
+  if (isVue2) {
+    // copy over global Vue APIs
+    options.router = componentInstance.$router;
```

Comment:
this can be put in the else of isVue3

Reply:
https://github.com/algolia/vue-instantsearch/pull/1020/commits/0a2a7784561b8dda4ed4ca4e165e1d652c598775